core/src/main/java/org/polypheny/db/adapter/DataStore.java

```diff
@@ -17,216 +17,63 @@
-     * support the document data model natively.
-     */
-    public void createCollection( Context prepareContext, CatalogCollection catalogCollection, long adapterId ) {
-        // overwrite this if the datastore supports document
-        createCollectionSubstitution( prepareContext, catalogCollection );
-    }
-
-
-    /**
-     * Substitution method, which is used to handle the {@link DataStore} required operations
-     * as if the data model would be {@link NamespaceType#RELATIONAL}.
-     */
-    private void createCollectionSubstitution( Context prepareContext, CatalogCollection catalogCollection ) {
-        Catalog catalog = Catalog.getInstance();
-        CatalogCollectionMapping mapping = catalog.getCollectionMapping( catalogCollection.id );
-
-        CatalogTable collectionEntity = catalog.getTable( mapping.collectionId );
-        createTable( prepareContext, collectionEntity, collectionEntity.partitionProperty.partitionIds );
-    }
-
-
-    /**
-     * Default method for dropping an existing collection on the {@link DataStore}.
-     * It comes with a substitution methods called by default and should be overwritten if the inheriting {@link DataStore}
-     * support the document data model natively.
-     */
-    public void dropCollection( Context prepareContext, CatalogCollection catalogCollection ) {
-        // overwrite this if the datastore supports document
-        dropCollectionSubstitution( prepareContext, catalogCollection );
-    }
-
-
-    /**
-     * Substitution method, which is used to handle the {@link DataStore} required operations
-     * as if the data model would be {@link NamespaceType#RELATIONAL}.
-     */
-    private void dropCollectionSubstitution( Context prepareContext, CatalogCollection catalogCollection ) {
-        Catalog catalog = Catalog.getInstance();
-        CatalogCollectionMapping mapping = catalog.getCollectionMapping( catalogCollection.id );
-
-        CatalogTable collectionEntity = catalog.getTable( mapping.collectionId );
-        dropTable( prepareContext, collectionEntity, collectionEntity.partitionProperty.partitionIds );
-    }
-
-
-    @AllArgsConstructor
-    public static class AvailableIndexMethod {
-
-        public final String name;
-        public final String displayName;
-
-    }
-
-
-    @AllArgsConstructor
-    public static class FunctionalIndexInfo {
-
-        public final List<Long> columnIds;
-        public final String methodDisplayName;
-
+    public record FunctionalIndexInfo(List<Long> columnIds, String methodDisplayName) {
 
         public List<String> getColumnNames() {
             List<String> columnNames = new ArrayList<>( columnIds.size() );
             for ( long columnId : columnIds ) {
-                columnNames.add( Catalog.getInstance().getColumn( columnId ).name );
+                // columnNames.add( Catalog.getInstance().getLogicalRel( columnNames ).getColumn( columnId ).name );
+                // todo dl
```

Comment:
This needs to be implemented or the method should be removed.

Reply:
Removed.